### PR TITLE
Remove texture reference completely

### PR DESCRIPTION
**Pull request by leofang**
Related to #7232 (see https://github.com/cupy/cupy/issues/7232#issuecomment-1347611986). This PR completely removes any APIs related to texture reference, which has been marked as deprecated since forever (?) ~~and is finally removed in CUDA 12+~~.

### CODE REVIEW

File: docs/source/user_guide/kernel.rst [hunk 1]
```diff
@@ -434,7 +434,7 @@ The instruction above for using complex numbers in :class:`~cupy.RawKernel` also
 
 For CUDA kernels that need to access global symbols, such as constant memory, the :meth:`~cupy.RawModule.get_global` method can be used, see its documentation for further detail.
 
-CuPy also supports the Texture Reference API. A handle to the texture reference in a module can be retrieved by name via :meth:`~cupy.RawModule.get_texref`. Then, you need to pass it to :class:`~cupy.cuda.texture.TextureReference`, along with a resource descriptor and texture descriptor, for binding the reference to the array. (The interface of :class:`~cupy.cuda.texture.TextureReference` is meant to mimic that of :class:`~cupy.cuda.texture.TextureObject` to help users make transition to the latter, since as of CUDA Toolkit 10.1 the former is marked as deprecated.)
+Note that the deprecated API :meth:`cupy.RawModule.get_texref` has been removed since CuPy vX.X due to the removal of texture reference support from CUDA.
```
Comment (leofang) [inline docs/source/user_guide/kernel.rst:437]: Once the release plan is decided, the version number here should be updated.

**Comment by leofang** [comment]
Sorry, I had wrong understanding. The texture reference APIs from CUDA **runtime** are removed in CUDA 12.0, but those from CUDA driver (that we use) are intact. Nevertheless, I still think it's good to be slated for removal in order to be in preparation for a future CUDA change and suppress build-time warnings. Functionality-wise, texture (and surface) objects are the way to go.

**Comment by leofang** [comment]
btw @kmaehashi @emcastillo this is technically a breaking change, but I seriously doubt there's any user... Just FYI for planning

**Comment by emcastillo** [comment]
yeah, we are not very sure about what to do with this and wanted to discuss it with you

**Comment by leofang** [comment]
That's what I figured 🙂 My justification is that there's no public project using `get_texref()`, which is the last API in the pipeline for utilizing texture references, so I assume the impact is very small:
https://github.com/search?q=cupy+get_texref+-org%3Acupy+language%3APython&type=code&l=Python

The other justification is we've documented it since the beginning that `TextureReference` is deprecated and subject to removal. Since all capabilities are also coveted by `TextureObject` and the `TextureReference` API was designed to mimic that of `TextureObject` (on purpose), I think even if there's user code broken it is at-most a one-line fix.

**Review by emcastillo — APPROVED** [review]
ByeBye

**Pull request merged**